ruby/example_code/s3/auth_federation_token_request_test.rb

```diff
@@ -1,142 +1,88 @@
-#**
- #* Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- #*
- #* This file is licensed under the Apache License, Version 2.0 (the "License").
- #* You may not use this file except in compliance with the License. A copy of
- #* the License is located at
- #*
- #* http://aws.amazon.com/apache2.0/
- #*
- #* This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
- #* CONDITIONS OF ANY KIND, either express or implied. See the License for the
- #* specific language governing permissions and limitations under the License.
-#**
-# snippet-sourcedescription:[auth_federation_token_request_test.rb allows a federated user with a limited set of permissions to lists keys in the specified bucket.] 
-# snippet-service:[s3]
-# snippet-keyword:[Ruby]
-# snippet-sourcesyntax:[ruby]
-# snippet-keyword:[Amazon S3]
-# snippet-keyword:[Code Sample]
-# snippet-keyword:[GET Bucket, GET Object, GET Keys]
-# snippet-sourcetype:[full-example]
-# snippet-sourcedate:[2019-02-11]
-# snippet-sourceauthor:[AWS]
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX - License - Identifier: Apache - 2.0
+
+# This code example allows a federated user with a limited set of
+# permissions to list the objects in the specified Amazon S3 bucket.
+
 # snippet-start:[s3.ruby.auth_federation_token_request_test.rb]
 require 'aws-sdk-s3'
 require 'aws-sdk-iam'
 
-USAGE = <<DOC
-
-Usage: ruby auth_federation_token_request_test.rb -b BUCKET -u USER [-r REGION] [-d] [-h]
-
-  Creates a federated policy for USER to list items in BUCKET for one hour.
-
-  BUCKET is required and must already exist.
-
-  USER is required and if not found, is created.
-
-  If REGION is not supplied, defaults to us-west-2.
-
-  -d gives you extra (debugging) information.
-
-  -h displays this message and quits.
-
-DOC
-
-def print_debug(debug, s)
-  if debug
-    puts s
-  end
-end
-
-# Get the user if they exist, otherwise create them
-def get_user(region, user_name, debug)
-  iam = Aws::IAM::Client.new(region: 'us-west-2')
-  
-  # See if user exists
-  user = iam.user(user_name)
- 
-  # If user does not exist, create them
-  if user == nil
-    user = iam.create_user(user_name: user_name)
-    iam.wait_until(:user_exists, user_name: user_name)
-    print_debug(debug, "Created new user #{user_name}")
-  else
-    print_debug(debug, "Found user #{user_name} in region #{region}")
+# Uses an AWS IAM user with a limited set of persmissions to list the objects
+# in the specified Amazon S3 bucket.
+# @param region [String] The AWS Region to use for configuring the AWS STS
+#   and S3 clients.
+# @param user_name [String] The user's name.
+# @param bucket_name [String] The bucket's name.
+# @return [Boolean] true if all operations succeed; otherwise, false.
```

Comment:
i would assume a method called `list_objects_with_temporary_credentials` would return a list of objects :) maybe either change this to return the objects or change the method name to say something like `can_list_objects_with_temporary_credentials?` since it returns true/false

Reply:
I changed this to can_...?, as you propose.